File: java/src/test/java/publicdomain/test/snippets/samples/RandomSamples.java
package publicdomain.test.snippets.samples;

import java.security.SecureRandom;

public class RandomSamples {
    private static final SecureRandom RAND = new SecureRandom();

    public static String aRandomName() {
        return someGivenName() + " Doe";
    }

    public static String someGivenName() {
        final String[] names = new String[]{"Alice", "Bob", "Carol", "David"};
        return names[aRandomIndexFor(names.length)];
    }

    public static int aRandomIndexFor(final int limit) {
        return (int) Math.abs(aRandomNumber() % (long) limit);
    }

    public static long aRandomNumber() {
        return RAND.nextLong();
    }
}

File: dotnet/TestSnippets.ForTest/Samples/RandomSamples.cs
using System;

namespace PublicDomain.TestSnippets.Samples
{
    public class RandomSamples
    {
        private static readonly Random Rand = new(Environment.TickCount);

        public static string ARandomName() {
            return SomeGivenName() + " Doe";
        }

        public static String SomeGivenName() {
            string[] names = {"Alice", "Bob", "Carol", "David" };
            return names[ARandomIndexFor(names.Length)];
        }
        
        public static int ARandomIndexFor(int limit) => Math.Abs(ARandomNumber() % limit);
        
        public static int ARandomNumber() => Rand.Next();
    }
}